app/services/discovery_engine/sync/locking.rb

```diff
@@ -49,6 +49,18 @@ def with_locked_document(content_id, payload_version:, &critical_section)
       raise FailedToAcquireLockError, "Failed to acquire lock for document: #{content_id}"
     end
 
+    # Determines if a given payload version is newer than the latest synced version for a document.
+    def newer_than_remote?(content_id, payload_version:)
+      # Consider a nil payload version as always newer
+      return true if payload_version.nil?
+
+      # If there is no remote version, our version is always newer by definition
+      remote_version = latest_synced_version(content_id)
+      return true if remote_version.nil?
```

Comment:
I think this check should come before the payload version check. If there's no remote version, then the current payload is always newer.

Reply:
See my comment for the other guard clause - the payload version check is really a sense check for an edge case, so I think it makes more sense for it to come first and the method to return straight away in that case, whereas this one is more of a "BAU" occurrence.